File: go/subdomain-visit-count.go
import (
    "strings"
    "strconv"
)

func AddSubdomains(subdomain_visit_count map[string]int, count int, domain string) {
    subdomain_visit_count[domain] += count
    for i := 0; i < len(domain); i++ {
        if domain[i] == '.' {
            subdomain := domain[i + 1:]
            subdomain_visit_count[subdomain] += count
        }
    }
}

func subdomainVisits(cpdomains []string) []string {
    subdomain_visit_count := make(map[string]int)
    for i := 0; i < len(cpdomains); i++ {
        entry := strings.Split(cpdomains[i], " ")
        count, _ := strconv.Atoi(entry[0])
        AddSubdomains(subdomain_visit_count, count, entry[1])
    }
    output := make([]string, 0)
    for subdomain, count := range subdomain_visit_count {
        entry := strconv.Itoa(count) + " " + subdomain
        output = append(output, entry)
    }
    return output
}

File: python/subdomain-visit-count.py
class Solution:
    def add_subdomains(domain: str, subdomain_visit_count:dict, count: int):
        subdomain_visit_count[domain] += count
        for i in range(len(domain)):
            if domain[i] == ".":
                subdomain_visit_count[domain[i + 1:]] += count

    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        subdomain_visit_count = defaultdict(lambda: 0)
        for entry in cpdomains:
            count, domain = entry.split()
            Solution.add_subdomains(domain, subdomain_visit_count, int(count))
        output = []
        for subdomain, count in subdomain_visit_count.items():
            output.append(f"{count} {subdomain}")
        return output
